test/tests/gcc-c-hello-world/run.sh

```diff
@@ -0,0 +1,10 @@
+#!/bin/bash
+set -e
+
+image="$1"
+dirTest="$(dirname "$(readlink -f "$BASH_SOURCE")")"
+dirContainer='/usr/src/c'
+
+docker run --rm -v "$dirTest":"$dirContainer":rw -w "$dirContainer" "$image" gcc -o hello-world  hello-world.c
+"$dirTest"/hello-world
+rm -rf "$dirTest"/hello-world
```

Comment:
That would also make it more feasible to run this test under Docker Machine on a non-Linux host (other incompatibilities notwithstanding).